### ISSUE TITLE
[Payment due 08-23][$250] Expensify card - Expensify card page content is reloaded upon navigating back

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#48172]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.25-4
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

**Issue found when executing PR** https://github.com/Expensify/App/pull/47844
## Action Performed:
1. Log in with an expensifail account
2. Create a workspace
3. Navigate to Workspace settings - More features
4. Enable "Expensify Card"
5. Navigate to Workspace settings - Expensify Card
6. Click on the "Issue new card" button
7. Click on the "<" button next to "Choose bank account"

## Expected Result:
It shouldn't reload

## Actual Result:
Expensify card page content is reloaded upon navigating back. On mWeb, it only reloads if you background the app and open it again

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/af68f564-4169-48fe-b9fb-b49b64562853

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01c51dda0396acf8a0</li>
        <li>Upwork Job ID: 1828835425848218614</li>
        <li>Last Price Increase: 2024-09-04</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @Christinadobrzyn</details>

**Comment by lanitochka17** [comment]
@Christinadobrzyn  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #wave-collect - Release 2

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Expensify card page reloads every time the page gets refocused.

### What is the root cause of that problem?
We have a `useFocusEffect` to fetch the card data.
https://github.com/Expensify/App/blob/1bb113191ddf05810e795e04f7651ddc89a58d5a/src/pages/workspace/expensifyCard/WorkspaceExpensifyCardPage.tsx#L30-L36

This is a common pattern within the workspace page, for example, in categories.
https://github.com/Expensify/App/blob/1bb113191ddf05810e795e04f7651ddc89a58d5a/src/pages/workspace/categories/WorkspaceCategoriesPage.tsx#L78-L82

The refetch will set the `cardSettings.isLoading` to true and based on the condition below, it will show the loading indicator (the cardSettings object only contains `isLoading` at this point).
https://github.com/Expensify/App/blob/1bb113191ddf05810e795e04f7651ddc89a58d5a/src/pages/workspace/expensifyCard/WorkspaceExpensifyCardPage.tsx#L47-L53
https://github.com/Expensify/App/blob/1bb113191ddf05810e795e04f7651ddc89a58d5a/src/pages/workspace/expensifyCard/WorkspaceExpensifyCardPage.tsx#L39

### What changes do you think we should make in order to solve the problem?

If we don't want to show the loading every time we refetch the data, we can change the condition to check whether the data exists or not.
```
const isLoading = !isOffline && (!cardSettings || !cardsList);
```

**Comment by Christinadobrzyn** [comment]
This is pretty minor, I don't know if we need to do this. @aimane-chnaif what do you think?

**Comment by Christinadobrzyn** [comment]
I'm going to be ooo 9/4-9/11.  @strepanier03 we are discussing if this needs to be fixed or if it should just be closed. 

@deetergp, can you check this issue out and let us know if you think we should fix it or close the issue. TY!

**Comment by deetergp** [comment]
It is minor, but not very premium. Since we've got an approved proposal, let's go for it!

**Comment by Christinadobrzyn** [comment]
Awesome! @aimane-chnaif can you review the provided proposals to see if any will work? TY!

**Comment by bernhardoj** [comment]
PR is ready

cc: @aimane-chnaif

**Comment by Christinadobrzyn** [comment]
monitoring PR - https://github.com/Expensify/App/pull/48613

**Comment by Christinadobrzyn** [comment]
PR in production today so setting payment due date.

**Comment by Christinadobrzyn** [comment]
Payment day- payment summary is here - https://github.com/Expensify/App/issues/48172#issuecomment-2354026204

Do we need a regression test?

**Comment by bernhardoj** [comment]
Yes, ND for me. Requested in ND.

**Comment by JmillsExpensify** [comment]
$250 approved for @bernhardoj

**Comment by Christinadobrzyn** [comment]
Thanks @aimane-chnaif and @bernhardoj! 

Regression test - https://github.com/Expensify/Expensify/issues/430723
@aimane-chnaif can you accept this offer - https://www.upwork.com/nx/wm/offer/104094994 TY!

**Comment by Christinadobrzyn** [comment]
I'm going to be ooo until 9/30- 

The only remaining piece is paying @aimane-chnaif in Upwork - https://www.upwork.com/nx/wm/offer/104094994 
payment summary - https://github.com/Expensify/App/issues/48172#issuecomment-2354026204

Everything else is done!

**Comment by abekkala** [comment]
@aimane-chnaif if you can accept the offer sent by @Christinadobrzyn I can get that paid out - thanks!

**Comment by Christinadobrzyn** [comment]
I'm back @abekkala - thanks for watching this! I'll reach out to @aimane-chnaif in a DM about payment.

**Comment by Christinadobrzyn** [comment]
I'm back @abekkala - thanks for watching this! I'll reach out to @aimane-chnaif in a DM about payment.

**Comment by Christinadobrzyn** [comment]
I'm gonna move this to weekly since we're just waiting on paying @aimane-chnaif in Upwork.

**Comment by Christinadobrzyn** [comment]
Paid @aimane-chnaif in Upwork. 

payment summary here - https://github.com/Expensify/App/issues/48172#issuecomment-2354026204

Closing as complete!
